TROUBLESHOOTING.md

```diff
@@ -0,0 +1,155 @@
+you must make some OS configuration.
+
+For your session:
+
+```shell script
+echo 1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+```
+
+Or definitively using `sysctl`:
+
+```shell script
+sudo sysctl -w kernel.perf_event_paranoid=1
+sudo sysctl -w kernel.kptr_restrict=0
+```
+
+For allocation profiling, you also need to install HotStop debug symbol.
+
+Depending on your Linux and Java distribution this can be done via:
+
+```shell script
+# Ubuntu/Debian - Java 8
+apt install openjdk-8-dbg
+
+# Ubuntu/Debian - Java 11
+ apt install openjdk-11-dbg
+
+# On CentOS, RHEL and some other RPM-based distributions
+debuginfo-install java-1.8.0-openjdk
+```
+
+If needed, see this section in the documentation site for details: [allocation profiling](https://github.com/jvm-profiling-tools/async-profiler#allocation-profiling).
+
+## Profiling application runtime with Async Profiler
+
+Asych Profiler comes with a Java agent, and a command line.
+
+To profile application during its runtime, I recommend using the command line as you can choose when to start the profiler, and avoid your profile to be bloated with startup events. <br/>
+Indeed, when an application start, it does a lot a bootstraping instruction, that will no more occur during the rest of the run of your application. <br/>
+By starting the profiling when needed, you avoid these bootstrap instructions to be part of the profile.
+
+When you profile an application runtime, it's good to create some load on your application using a load generator tool (wrk, gatling, ...),
+and to start profiling only after some warmup time to allow the Just In Time compiler of Java to optimize your application code (and your database cache to warmup, etc...).
+
+### CPU profiling
+
+To start CPU profiling, enter the following command:
+
+```shell script
+./profiler.sh start -b 4000000 <pid>
+```
+
+`-b 4000000` is used to increase the frame buffer size as the default is usually too small.
+
+To end profiling and gathering the result you can launch the same command with the `stop` subcommand, this will tells you if the buffer frame was too small. <br/>
+The output is a text file that is not really usable, so let's use our preferred performance representation: The FlameGraph.
+
+```shell script
+./profiler.sh stop -f /tmp/cpu-profile.svg --width 1600 <pid>
+```
+
+It will create an SVG FlameGraph (Async Profiler automatically detect that you ask for a FlameGraph thanks to the svg extension)
+that you can open in your browser (and even zoom inside it by clicking on a frame). <br/>
+The example command will create an SVG of 1600 pixels width.
+
+One of our favorite options is `-s` (or `--simple`) that use simple class names instead of FQDNs, the FlameGraph will be more readable with it, but you will lose the package name. <br/>
+You can also limit the profiling duration by using `-d` (or `--duration`) follow by the duration in seconds.
```

Comment:
```suggestion
You can also limit the profiling duration by using `-d` (or `--duration`) follow by the duration in seconds. If you use the `--duration` option, the output file will be created automatically at the end of the duration period. You do not need to start and stop the profiler.
```

Reply:
I think we still need to start the profiler so it should be 

> You do not need to stop the profiler.

Reply:
If you execute ;
```
$ ./profiler.sh -d 30 <pid>
```
This is the same as;
```
$ ./profiler.sh start <pid>
$ sleep 30
$ ./profiler.sh stop <pid>
```
This will attach the profiler to process `<pid>`, run for 30s then stop and write the output. A user does not need to execute `$ ./profiler.sh start -b <pid>` beforehand

Maybe this should be clearer, e.g. "You do not need to explicitly start and stop the profiler."

Reply:
Ah OK, so start is the default command!

Reply:
Done